pyomo/common/config.py

```diff
@@ -1650,85 +1750,97 @@ def __setstate__(self, state):
         self._userAccessed = True
-        key = str(key)
-        if isinstance(self._data[key], ConfigValue):
-            return self._data[key].value()
+        _key = str(key).replace(' ','_')
+        if isinstance(self._data[_key], ConfigValue):
+            return self._data[_key].value()
         else:
-            return self._data[key]
+            return self._data[_key]
 
-    def get(self, key, default=ConfigBase.NoArgument):
+    def get(self, key, default=NoArgumentGiven):
         self._userAccessed = True
-        key = str(key)
-        if key in self._data:
-            return self._data[key]
-        if default is ConfigBase.NoArgument:
+        _key = str(key).replace(' ','_')
+        if _key in self._data:
+            return self._data[_key]
+        if default is NoArgumentGiven:
             return None
         if self._implicit_domain is not None:
             return self._implicit_domain(default)
         else:
             return ConfigValue(default)
 
-    def setdefault(self, key, default=ConfigBase.NoArgument):
+    def setdefault(self, key, default=NoArgumentGiven):
         self._userAccessed = True
-        key = str(key)
-        if key in self._data:
-            return self._data[key]
-        if default is ConfigBase.NoArgument:
+        _key = str(key).replace(' ','_')
+        if _key in self._data:
+            return self._data[_key]
+        if default is NoArgumentGiven:
             return self.add(key, None)
         else:
             return self.add(key, default)
 
     def __setitem__(self, key, val):
-        key = str(key)
-        if key not in self._data:
+        _key = str(key).replace(' ','_')
+        if _key not in self._data:
             self.add(key, val)
         else:
-            self._data[key].set_value(val)
+            self._data[_key].set_value(val)
         #self._userAccessed = True
 
     def __delitem__(self, key):
         # Note that this will produce a KeyError if the key is not valid
         # for this ConfigDict.
-        del self._data[key]
+        _key = str(key).replace(' ','_')
+        del self._data[_key]
         # Clean up the other data structures
-        self._decl_order.remove(key)
-        self._declared.discard(key)
+        self._decl_order.remove(_key)
+        self._declared.discard(_key)
 
     def __contains__(self, key):
-        key = str(key)
-        return key in self._data
+        _key = str(key).replace(' ','_')
+        return _key in self._data
 
     def __len__(self):
         return self._decl_order.__len__()
 
     def __iter__(self):
-        return self._decl_order.__iter__()
+        return (self._data[key]._name for key in self._decl_order)
 
     def __getattr__(self, name):
         # Note: __getattr__ is only called after all "usual" attribute
         # lookup methods have failed.  So, if we get here, we already
         # know that key is not a __slot__ or a method, etc...
         #if name in ConfigDict._all_slots:
```

Comment:
Is this commented out line still necessary?

Reply:
Not strictly necessary, but I found it useful to job my memory when I was returning to this code.